### Load packages and dataset

In [1]:
# import packages

import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
import math

In [2]:
# load dataset

comm_data = pd.read_csv("datasets/bureau_survey.csv")
display(comm_data.head())

RT  SERIALNO  DIVISION  PUMA  REGION  ST   ADJHSG   ADJINC  WGTP  NP  \
0  H        37         6  1000       3   1  1000000  1001264    69   0   
1  H        67         6  2701       3   1  1000000  1001264    48   4   
2  H        90         6   400       3   1  1000000  1001264    66   0   
3  H       160         6  1000       3   1  1000000  1001264    65   1   
4  H       345         6  1000       3   1  1000000  1001264   148   1   

    ...    wgtp71  wgtp72  wgtp73  wgtp74  wgtp75  wgtp76  wgtp77  wgtp78  \
0   ...       126      63      56     100     126      67     136     110   
1   ...        40      50      56      44      59      82      50      16   
2   ...        78     109      57      90     100      25      75      19   
3   ...        78     135      62      74     100      15      64      47   
4   ...        46      49     168     141     142     180     138     287   

   wgtp79  wgtp80  
0      69      19  
1      67      70  
2      59      75  
3      23      21  
4     260      41  

[5 rows x 235 columns]

## Task 1

In [4]:
comm_data

RT  SERIALNO  DIVISION  PUMA  REGION  ST   ADJHSG   ADJINC  WGTP  NP  \
0      H        37         6  1000       3   1  1000000  1001264    69   0   
1      H        67         6  2701       3   1  1000000  1001264    48   4   
2      H        90         6   400       3   1  1000000  1001264    66   0   
3      H       160         6  1000       3   1  1000000  1001264    65   1   
4      H       345         6  1000       3   1  1000000  1001264   148   1   
5      H       463         6  2701       3   1  1000000  1001264    36   3   
6      H       477         6  1100       3   1  1000000  1001264    60   1   
7      H       518         6   100       3   1  1000000  1001264   175   4   
8      H       535         6  1000       3   1  1000000  1001264    89   2   
9      H       566         6   500       3   1  1000000  1001264    12   1   
10     H       717         6  1900       3   1  1000000  1001264   119   4   
11     H       806         6   302       3   1  1000000  1001264    90   1   
12     H       810         6  2500       3   1  1000000  1001264    12   4   
13     H       848         6  1301       3   1  1000000  1001264   170   1   
14     H      1014         6  1304       3   1  1000000  1001264   111   3   
15     H      1052         6  2600       3   1  1000000  1001264     0   1   
16     H      1150         6  2200       3   1  1000000  1001264    90   0   
17     H      1160         6  1305       3   1  1000000  1001264    79   5   
18     H      1215         6  2500       3   1  1000000  1001264    16   1   
19     H      1346         6  2200       3   1  1000000  1001264    40   0   
20     H      1471         6  2701       3   1  1000000  1001264    87   0   
21     H      1563         6  2703       3   1  1000000  1001264    92   1   
22     H      1610         6  2702       3   1  1000000  1001264   277   1   
23     H      1619         6  1500       3   1  1000000  1001264   112   4   
24     H      1685         6  1400       3   1  1000000  1001264    73   1   
25     H      1689         6  1304       3   1  1000000  1001264    33   2   
26     H      1737         6  2400       3   1  1000000  1001264   123   0   
27     H      1871         6  2500       3   1  1000000  1001264    26   2   
28     H      1890         6  1900       3   1  1000000  1001264     0   1   
29     H      1898         6   600       3   1  1000000  1001264   112   3   
...   ..       ...       ...   ...     ...  ..      ...      ...   ...  ..   
19101  H   1180128         6  1400       3   1  1000000  1001264    77   2   
19102  H   1180172         6  2200       3   1  1000000  1001264    48   2   
19103  H   1180259         6  1303       3   1  1000000  1001264    39   1   
19104  H   1180280         6  2701       3   1  1000000  1001264    58   3   
19105  H   1180366         6  1100       3   1  1000000  1001264   271   3   
19106  H   1180382         6   200       3   1  1000000  1001264     0   1   
19107  H   1180384         6  2100       3   1  1000000  1001264     0   1   
19108  H   1180642         6   700       3   1  1000000  1001264   138   2   
19109  H   1180738         6   302       3   1  1000000  1001264   100   1   
19110  H   1180747         6  2000       3   1  1000000  1001264   190   4   
19111  H   1180830         6   500       3   1  1000000  1001264    35   2   
19112  H   1180904         6   301       3   1  1000000  1001264   140   1   
19113  H   1180954         6  2500       3   1  1000000  1001264    42   1   
19114  H   1180979         6  1900       3   1  1000000  1001264     0   1   
19115  H   1181005         6  1400       3   1  1000000  1001264   111   0   
19116  H   1181022         6   100       3   1  1000000  1001264    82   1   
19117  H   1181033         6  2200       3   1  1000000  1001264    89   2   
19118  H   1181041         6   400       3   1  1000000  1001264   210   2   
19119  H   1181066         6  1600       3   1  1000000  1001264   325   0   
19120  H   1181083         6

In [3]:
# create a sample

size = 10000 #sample size
comm_data_sample = comm_data.sample(size, random_state=0)
display(comm_data_sample.head())

RT  SERIALNO  DIVISION  PUMA  REGION  ST   ADJHSG   ADJINC  WGTP  NP  \
6344  H    398761         6  2400       3   1  1000000  1001264    12   2   
7251  H    455027         6   200       3   1  1000000  1001264   422   1   
1992  H    126087         6  1000       3   1  1000000  1001264    21   0   
998   H     63339         6  1303       3   1  1000000  1001264   166   2   
4025  H    250466         6  1303       3   1  1000000  1001264   125   4   

       ...    wgtp71  wgtp72  wgtp73  wgtp74  wgtp75  wgtp76  wgtp77  wgtp78  \
6344   ...        16      12      27      19       3      10       2      15   
7251   ...       696     376     516     480     196     466     151     751   
1992   ...         5      15      32       7       8      29      21      18   
998    ...        59     153     154     280     261      54     165     182   
4025   ...       184     139     131     120     137     115      37      43   

      wgtp79  wgtp80  
6344      11      23  
7251     383     485  
1992      35      33  
998      163     279  
4025      33     228  

[5 rows x 235 columns]

<p>
<details><summary>Click to show solution</summary><br>
    
```python
# create a sample

size = 100000 #sample size
comm_data_sample = comm_data.sample(size, random_state=0)
display(comm_data_sample.head())
```

</details>
</p>

##  Task 2

In [5]:
# calculate the sample proportion of households with access to the internet

n = comm_data_sample.shape[0] #sample size

count_access = 0 #counter to keep track of no. households with internet access

for i in comm_data_sample.index:
    
    if comm_data_sample.loc[i, "ACCESS"] == 1 or comm_data_sample.loc[i, "ACCESS"] == 2:
        #this household has internet access
        #increment the counter
        count_access = count_access + 1

# calculate sample proportion
# p_hat =(no. households with internet access/total no. households)
p_hat = count_access/n

print("Sample proportion of households with access to the internet:", p_hat)

Sample proportion of households with access to the internet: 0.5884


In [6]:
# generate 95% confidence interval for population proportion

low_ci_bound = p_hat - 1.96 * math.sqrt(p_hat * (1-p_hat) /n)
high_ci_bound = p_hat + 1.96 * math.sqrt(p_hat * (1-p_hat) /n)

prop_ci = [low_ci_bound, high_ci_bound]
print("95% confidence interval for population proportion:", prop_ci)

95% confidence interval for population proportion: [0.5787543813764798, 0.5980456186235202]


<p>
<details><summary>Click to show answer</summary><br>
    
`95% confidence interval for population proportion: [0.5787543813764798, 0.5980456186235202]`

</details>
</p>

<p>
<details><summary>Click to show solution</summary><br>
    
```python
# generate 95% confidence interval for population proportion

low_ci_bound = p_hat - 1.96*(math.sqrt((p_hat*(1-p_hat))/(n)))
high_ci_bound = p_hat + 1.96*(math.sqrt((p_hat*(1-p_hat))/(n)))

prop_ci = [low_ci_bound, high_ci_bound]
print("95% confidence interval for population proportion:", prop_ci)
```

</details>
</p>

## Task 3

In [7]:
# generate 95% confidence interval for population mean

# n is the sample size

all_num_people_values = comm_data_sample["NP"]
num_people_mean = np.mean(all_num_people_values)
num_people_stdev = np.std(all_num_people_values)

low_ci_bound = num_people_mean - 1.96 * num_people_stdev / math.sqrt(n)
high_ci_bound = num_people_mean + 1.96 * num_people_stdev / math.sqrt(n)

mean_ci = [low_ci_bound, high_ci_bound]
print("95% confidence interval for population mean:", mean_ci)

95% confidence interval for population mean: [1.907986823337287, 1.9618131766627132]


In [14]:
all_num_people_values

6344     2
7251     1
1992     0
998      2
4025     4
14091    2
7958     3
10341    6
2156     1
18262    3
9957     3
2438     1
14868    2
9086     1
8494     1
11891    2
1801     2
17789    0
14939    1
8011     1
16815    1
11853    0
7986     2
1751     1
12502    2
6164     0
9884     6
12116    2
17059    1
3479     2
        ..
2106     2
9360     2
13233    2
4113     1
11841    3
13588    0
5857     2
7199     3
11153    1
13525    1
6849     1
11110    2
8607     2
17777    1
13109    3
3758     3
12076    0
1411     4
17447    1
8552     1
8929     2
5281     2
16296    1
11627    1
4738     1
5576     1
14754    1
14035    2
12677    2
3482     1
Name: NP, Length: 10000, dtype: int64

<p>
<details><summary>Click to show answer</summary><br>
    
`95% confidence interval for population mean: [1.907986823337287, 1.9618131766627132]`

</details>
</p>

<p>  
<details><summary>Click to show solution</summary><br>
    
```python
# generate 95% confidence interval for population mean

all_num_people_values = comm_data_sample["NP"]
num_people_mean = np.mean(all_num_people_values)
num_people_stdev = np.std(all_num_people_values)

low_ci_bound = num_people_mean - 1.96*(num_people_stdev/math.sqrt(n))
high_ci_bound = num_people_mean + 1.96*(num_people_stdev/math.sqrt(n))

mean_ci = [low_ci_bound, high_ci_bound]
print("95% confidence interval for population mean:", mean_ci)
```

</details>
</p>

## Task 4

In [8]:
# create a multigenerational household sample and a non-multigenerational household sample

multi_gen = comm_data[comm_data["MULTG"] == 2]
non_multi_gen = comm_data[comm_data["MULTG"] == 1]
print("Multigenerational household sample:")
display(multi_gen.head())
print("Non-multigenerational household sample:")
display(non_multi_gen.head())

Multigenerational household sample:


RT  SERIALNO  DIVISION  PUMA  REGION  ST   ADJHSG   ADJINC  WGTP  NP  \
12   H       810         6  2500       3   1  1000000  1001264    12   4   
44   H      3589         6   700       3   1  1000000  1001264    78   5   
191  H     13338         6  2300       3   1  1000000  1001264    27   8   
198  H     13673         6  1700       3   1  1000000  1001264    41   3   
221  H     15234         6  1200       3   1  1000000  1001264   222   4   

      ...    wgtp71  wgtp72  wgtp73  wgtp74  wgtp75  wgtp76  wgtp77  wgtp78  \
12    ...         4      14       5      12      14       4      12       3   
44    ...        32      25      19      63      98      88      84      71   
191   ...        30      29      20       8      26      43       9      28   
198   ...        26       7      41      18      39      31      31      65   
221   ...        65     202     406     375     342     237     227     188   

     wgtp79  wgtp80  
12       22      12  
44      105     148  
191       8      42  
198      47      14  
221     200     368  

[5 rows x 235 columns]

Non-multigenerational household sample:


RT  SERIALNO  DIVISION  PUMA  REGION  ST   ADJHSG   ADJINC  WGTP  NP  \
1  H        67         6  2701       3   1  1000000  1001264    48   4   
3  H       160         6  1000       3   1  1000000  1001264    65   1   
4  H       345         6  1000       3   1  1000000  1001264   148   1   
5  H       463         6  2701       3   1  1000000  1001264    36   3   
6  H       477         6  1100       3   1  1000000  1001264    60   1   

    ...    wgtp71  wgtp72  wgtp73  wgtp74  wgtp75  wgtp76  wgtp77  wgtp78  \
1   ...        40      50      56      44      59      82      50      16   
3   ...        78     135      62      74     100      15      64      47   
4   ...        46      49     168     141     142     180     138     287   
5   ...        65       9      34      10      41      40      37      14   
6   ...        59      72      89      20      84      60      80      78   

   wgtp79  wgtp80  
1      67      70  
3      23      21  
4     260      41  
5      11      11  
6      98      47  

[5 rows x 235 columns]

In [10]:
# calculate the mean and stdev for each sample

multi_gen_np_vals = multi_gen["NP"]
multi_gen_mean = np.mean(multi_gen_np_vals)
multi_gen_stdev = np.std(multi_gen_np_vals)

non_multi_gen_np_vals = non_multi_gen["NP"]
non_multi_gen_mean = np.mean(non_multi_gen_np_vals)
non_multi_gen_stdev =  np.std(non_multi_gen_np_vals)

print("Mean number of people in multigenerational households:", multi_gen_mean)
print("Standard deviation for number of people in multigenerational households:", multi_gen_stdev)

print("Mean number of people in non-multigenerational households:", non_multi_gen_mean)
print("Standard deviation for number of people in non-multigenerational households:", non_multi_gen_stdev)

Mean number of people in multigenerational households: 4.658135283363802
Standard deviation for number of people in multigenerational households: 1.3032593776621781
Mean number of people in non-multigenerational households: 2.2074784017278617
Standard deviation for number of people in non-multigenerational households: 1.196315082012862


<p>
<details><summary>Click to show answer</summary><br>
    
`Mean number of people in multigenerational households: 4.658135283363802`<br>
`Standard deviation for number of people in multigenerational households: 1.3032593776621781`<br>
`Mean number of people in non-multigenerational households: 2.2074784017278617`<br>
`Standard deviation for number of people in non-multigenerational households: 1.196315082012862`

</details>
</p>

<p>
<details><summary>Click to show solution</summary><br>
    
```python
# calculate the mean and stdev for each sample

multi_gen_np_vals = multi_gen["NP"]
multi_gen_mean = np.mean(multi_gen_np_vals)
multi_gen_stdev = np.std(multi_gen_np_vals)

non_multi_gen_np_vals = non_multi_gen["NP"]
non_multi_gen_mean = np.mean(non_multi_gen_np_vals)
non_multi_gen_stdev = np.std(non_multi_gen_np_vals)

print("Mean number of people in multigenerational households:", multi_gen_mean)
print("Standard deviation for number of people in multigenerational households:", multi_gen_stdev)

print("Mean number of people in non-multigenerational households:", non_multi_gen_mean)
print("Standard deviation for number of people in non-multigenerational households:", non_multi_gen_stdev)
```

</details>
</p>

In [13]:
# generate 95% confidence interval for population mean difference

n1 = multi_gen.shape[0]
n2 = non_multi_gen.shape[0]

low_ci_bound = (multi_gen_mean - non_multi_gen_mean) - 1.96*(
    math.sqrt((multi_gen_stdev**2)/n1 + (non_multi_gen_stdev**2)/n2))
high_ci_bound = (multi_gen_mean - non_multi_gen_mean) + 1.96*(
    math.sqrt((multi_gen_stdev**2)/n1 + (non_multi_gen_stdev**2)/n2))

mean_diff_ci = [low_ci_bound, high_ci_bound]
print("95% confidence interval for population mean difference:", mean_diff_ci)

95% confidence interval for population mean difference: [2.3397533051583563, 2.561560458113525]


In [11]:
multi_gen_mean - non_multi_gen_mean

2.4506568816359406

In [12]:
3 **2

9

<p>
<details><summary>Click to show answer</summary><br>
    
`95% confidence interval for population mean difference: [2.3397533051583563, 2.561560458113525]`

</details>
</p>

<p>
<details><summary>Click to show solution</summary><br>
    
```python
# generate 95% confidence interval for population mean difference

# note that the \ symbol used below just splits code into separate lines for easier reading

n1 = multi_gen.shape[0]
n2 = non_multi_gen.shape[0]

low_ci_bound = (multi_gen_mean - non_multi_gen_mean) \
                - 1.96*(math.sqrt((multi_gen_stdev**2)/n1 + (non_multi_gen_stdev**2)/n2))
high_ci_bound = (multi_gen_mean - non_multi_gen_mean) \
                + 1.96*(math.sqrt((multi_gen_stdev**2)/n1 + (non_multi_gen_stdev**2)/n2))

mean_diff_ci = [low_ci_bound, high_ci_bound]
print("95% confidence interval for population mean difference:", mean_diff_ci)
```

</details>
</p>